From: "Haidong Yu" <yhd986@163.com>
#Subject: [PATCH] user32: The parent window with WS_CLIPCHILDREN style should not be erased when hiding or resizing child.
Message-Id: <39ed3b3b.56dd.1784625772d.Coremail.yhd986@163.com>
Date: Fri, 19 Mar 2021 00:22:43 +0800 (CST)

<br><br><br><br><span title="neteasefooter"><p>&nbsp;</p></span>

From 080e406ea7729c64f420b24c6bf8accf52f30362 Mon Sep 17 00:00:00 2001
From: Haidong Yu <yuhaidong@uniontech.com>
Date: Thu, 18 Mar 2021 23:20:40 +0800
Subject: [PATCH] user32: The parent window with WS_CLIPCHILDREN style should
 not be erased when hiding or resizing child.

Signed-off-by: Haidong Yu <yuhaidong@uniontech.com>
---
 dlls/user32/winpos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/user32/winpos.c b/dlls/user32/winpos.c
index c6f806f7beb..e87c5c3eef3 100644
--- a/dlls/user32/winpos.c
+++ b/dlls/user32/winpos.c
@@ -2337,7 +2337,7 @@ BOOL USER_SetWindowPos( WINDOWPOS * winpos, int parent_x, int parent_y )
         {
             HWND parent = GetAncestor( winpos->hwnd, GA_PARENT );
             if (!parent || parent == GetDesktopWindow()) parent = winpos->hwnd;
-            erase_now( parent, 0 );
+            if (!(GetWindowLongW(parent, GWL_STYLE) & WS_CLIPCHILDREN)) erase_now( parent, 0 );
         }
 
         /* Give newly shown windows a chance to redraw */

-- 
2.20.1

